### DISCUSSION TITLE
An  AttributeError occurred while running the code about pore diameter

### DISCUSSION THREAD

**Discussion by Devin-one** [PMEAL/OpenPNM#2400] [Q&A]
Hi, Professor. I asked you a question on the problem of functional expression of porosity [#2369](url)  the other day.  But when I was running, I made the following mistakes.
<img width="686" alt="78ee76b48fb7353775e8764a76028af" src="https://user-images.githubusercontent.com/93040662/166152434-606ef3f5-034d-4f96-9c95-5ca9e73a6445.png">
I wonder if my code is wrong. By the way, if I calculate the porosity and permeability in two dimensions, it doesn't matter whether Lz is equal to 1 in the cubic network, does it？

**Comment by jgostick** [comment]
It's because you're assigning a numpy array to 'pore.seed', then passing 'pore.seed' in as the model.  Openpnn is treating the numpy array as a function, but it's not a function.  Judging by your code above it looks like you just want to assign im1 to 'pore.seed' then you can drop the add_model call and just go straight to the 'pore.diameter' model.
